### ISSUE TITLE
Build failed with 2018-12-26 nightly

### ISSUE THREAD

**Issue by Riey** [rust-lang-nursery/futures-rs#1374]
```
error[E0599]: no function or associated item named `get_mut_unchecked` found for type `std::pin::Pin<_>` in the current scope
   --> /home/riey/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-core-preview-0.3.0-alpha.10/src/future/future_obj.rs:176:40
    |
176 |         let mut_ref: &mut F = unsafe { Pin::get_mut_unchecked(Pin::as_mut(&mut self)) };
    |                                        ^^^^^^^^^^^^^^^^^^^^^^ function or associated item not found in `std::pin::Pin<_>`

error[E0599]: no function or associated item named `get_mut_unchecked` found for type `std::pin::Pin<_>` in the current scope
   --> /home/riey/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-core-preview-0.3.0-alpha.10/src/future/future_obj.rs:216:44
    |
216 |             let mut_ref: &mut F = unsafe { Pin::get_mut_unchecked(Pin::as_mut(&mut self)) };
    |                                            ^^^^^^^^^^^^^^^^^^^^^^ function or associated item not found in `std::pin::Pin<_>`

error[E0277]: the trait bound `P: std::marker::Unpin` is not satisfied in `std::pin::Pin<P>`
  --> /home/riey/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-core-preview-0.3.0-alpha.10/src/stream/mod.rs:83:9
   |
83 |         Pin::get_mut(self).as_mut().poll_next(lw)
   |         ^^^^^^^^^^^^ within `std::pin::Pin<P>`, the trait `std::marker::Unpin` is not implemented for `P`
   |
   = help: consider adding a `where P: std::marker::Unpin` bound
   = note: required because it appears within the type `std::pin::Pin<P>`
   = note: required by `<std::pin::Pin<&'a mut T>>::get_mut`

error[E0599]: no function or associated item named `get_mut_unchecked` found for type `std::pin::Pin<_>` in the current scope
  --> /home/riey/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-core-preview-0.3.0-alpha.10/src/stream/mod.rs:96:19
   |
96 |             match Pin::get_mut_unchecked(self) {
   |                   ^^^^^^^^^^^^^^^^^^^^^^ function or associated item not found in `std::pin::Pin<_>`

error[E0599]: no method named `poll_next` found for type `std::pin::Pin<_>` in the current scope
  --> /home/riey/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-core-preview-0.3.0-alpha.10/src/stream/mod.rs:97:58
   |
97 |                 Either::Left(a) => Pin::new_unchecked(a).poll_next(lw),
   |                                                          ^^^^^^^^^
   |
   = note: Pin::new_unchecked(a) is a function, perhaps you wish to call it
   = help: items from traits can only be used if the trait is implemented and in scope
   = note: the following traits define an item `poll_next`, perhaps you need to implement one of them:
           candidate #1: `stream::stream_obj::UnsafeStreamObj`
           candidate #2: `stream::Stream`

error[E0599]: no method named `poll_next` found for type `std::pin::Pin<_>` in the current scope
  --> /home/riey/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-core-preview-0.3.0-alpha.10/src/stream/mod.rs:98:59
   |
98 |                 Either::Right(b) => Pin::new_unchecked(b).poll_next(lw),
   |                                                           ^^^^^^^^^
   |
   = note: Pin::new_unchecked(b) is a function, perhaps you wish to call it
   = help: items from traits can only be used if the trait is implemented and in scope
   = note: the following traits define an item `poll_next`, perhaps you need to implement one of them:
           candidate #1: `stream::stream_obj::UnsafeStreamObj`
           candidate #2: `stream::Stream`

error[E0599]: no function or associated item named `get_mut_unchecked` found for type `std::pin::Pin<_>` in the current scope
   --> /home/riey/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-core-preview-0.3.0-alpha.10/src/stream/stream_obj.rs:178:18
    |
178 |         unsafe { Pin::get_mut_unchecked(self) as *mut F as *mut () }
    |                  ^^^^^^^^^^^^^^^^^^^^^^ function or associated item not found in `std::pin::Pin<_>`

error[E0599]: no function or associated item named `get_mut_unchecked` found for type `std::pin::Pin<_>` in the current scope
   --> /home/riey/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-core-preview-0.3.0-alpha.10/src/stream/stream_obj.rs:218:44
    |
218 |             let mut_ref: &mut F = unsafe { Pin::get_mut_unchecked(Pin::as_mut(&mut self)) };
    |                                            ^^^^^^^^^^^^^^^^^^^^^^ function or associated item not found in `std::pin::Pin<_>`
```

I think this is because the std's api changed in current version

e.g. https://doc.rust-lang.org/nightly/src/core/pin.rs.html#258-260

and 2018-12-24 version of nightly works good

**Comment by zippy** [comment]
For me the recent update of pin-utils is breaking my use of futures-previe-0.3.0-alpha.10 on any nightly that I use 26.  Over on [holochain-rust](https://github.com/holochain/holochain-rust) we are currently pinning our dev work to 2018-11-28 and as of this morning trying compile fails with:

```
error[E0599]: no function or associated item named `get_unchecked_mut` found for type `std::pin::Pin<_>` in the current scope
  --> /home/eric/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-util-preview-0.3.0-alpha.10/src/future/map.rs:19:5
   |
19 |     unsafe_unpinned!(f: Option<F>);
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ function or associated item not found in `std::pin::Pin<_>`
   |
This seems 
   = help: did you mean `map_unchecked_mut`?
   = note: this error originates in a macro outside of the current crate (in Nightly builds, run with -Z external-macro-backtrace for more info)

error[E0599]: no function or associated item named `get_unchecked_mut` found for type `std::pin::Pin<_>` in the current scope
  --> /home/eric/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-util-preview-0.3.0-alpha.10/src/future/inspect.rs:19:5
   |
19 |     unsafe_unpinned!(f: Option<F>);
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ function or associated item not found in `std::pin::Pin<_>`
   |
   = help: did you mean `map_unchecked_mut`?
   = note: this error originates in a macro outside of the current crate (in Nightly builds, run with -Z external-macro-backtrace for more info)

... many more...
```
It looks like this is actually related to the recent upgrade of [pin-utils](https://github.com/rust-lang-nursery/pin-utils) to alpha.4.  But this seems very strange to me because I see that futures-preview uses alpha.3.  So I don't understand why code would be breaking downstream that's setting its dependency to  `futures-preview = "0.3.0-alpha.10"` which should still be using alpha.3 of pin-utils.  Any help here?

**Comment by Riey** [comment]
Fixed by #1377
